benchmark_runner/common/google_drive/google_drive_operations.py

```diff
@@ -241,16 +241,38 @@ def get_drive_folder_url(self, folder_path, parent_folder_id):
             # Folder exists, return the folder URL
             folder_url = f"{self._google_drive_path}/{folder_id}"
             return folder_url
-        else:
-            # If folder does not exist, create it along the path
-            folder_id = self.create_folder_at_path(folder_path, parent_folder_id)
+
+    def find_or_create_drive_folder_url(self, folder_path, parent_folder_id, retries=3, delay=5):
+        """
+        Find an existing folder by path or create it. Retries in case of failure.
+        :param folder_path: The full folder path to retrieve or create.
+        :param parent_folder_id: The starting parent folder ID.
+        :param retries: Number of retry attempts on failure.
+        :param delay: Initial delay between retries in seconds.
+        :return: The Google Drive URL for the folder, or None on failure.
+        """
+        for attempt in range(1, retries + 1):
+            folder_id = self.get_folder_id_by_path(folder_path, parent_folder_id)
```

Comment:
The race condition still exists; the check for it could fail, but something else could create it before we get around to it.  It's likely a rather narrow race condition, but I'd be very surprised if Google Drive does not offer a way to create a folder only if it does not exist.  Please try to look for that way.

Reply:
I tested it and it's working properly. I call it once before all the runs, so it's created only once, and that solved the issue.

Reply:
It will work properly if you can guarantee that only one instance is running at a time.  The problem is if multiple instances are running and both want to create the same folder name.  As I said, a small window, but it's not a great idea to assume that we're the only use case.